duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -175,299 +205,480 @@ namespace caf {
           }
         }
+
+        SRDirectionBranch &dir = reco.dir;
+        FillDirectionInfo(dir, evt);
+
+
+        //Neutrino flavours hypotheses
+        SRNeutrinoHypothesisBranch &nuhyp = reco.nuhyp;
+        //Filling only CVN at the moment. Filling the same info for all the particles -> CVN applies to the whole event as far as I know
+        FillCVNInfo(nuhyp.cvn, evt);
+
+        //Neutrino energy hypothese
+        SRNeutrinoEnergyBranch &Enu = reco.Enu;
+        FillEnergyInfo(Enu, evt);
+
+        //List of reconstructed particles
+        SRRecoParticlesBranch &part = reco.part;
+        FillRecoParticlesInfo(part, evt);
+
+        reco.truth = {0}; //Assuming a single TrueInteraction for now
+        //TODO, not sure of what to put there..
+        // std::vector<float>   truthOverlap;              ///< Fractional overlap between this reco interaction and each true interaction
+
+        pandora.emplace_back(reco);
       }
-    } // loop through MC truth i
+    }
+
+    ixn.npandora = pandora.size();
+    ixn.ndlp = ixn.dlp.size();
   }
 
+
   //------------------------------------------------------------------------------
+ 
+ 
   void CAFMaker::beginSubRun(const art::SubRun& sr)
   {
-    auto pots = sr.getHandle< sumdata::POTSummary >("generator");
-    if( pots ) meta_pot += pots->totpot;
+    art::Handle<sumdata::POTSummary> pots = sr.getHandle<sumdata::POTSummary>("generator");
+    if( pots ) fMetaPOT += pots->totpot;
+
+    fMetaRun = sr.id().subRun();
+    fMetaSubRun = sr.id().run();
+
   }
 
   //------------------------------------------------------------------------------
-  void CAFMaker::analyze(art::Event const & evt)
+
+  void CAFMaker::FillMetaInfo(caf::SRDetectorMeta &meta, const art::Event &evt) const
   {
-    caf::StandardRecord sr;
+    meta.enabled = true;
+    meta.run = evt.id().run();
+    meta.subrun = evt.id().subRun();
+    meta.event = evt.id().event();
+    meta.subevt = 0; //Hardcoded to 0, not sure of how this should be used
 
-    if(fTree){
-      caf::StandardRecord* psr = &sr;
-      fTree->SetBranchAddress("rec", &psr);
-    }
+    //Nothing is filled about the trigger for the moment
+  }
+
+  //------------------------------------------------------------------------------
+
+  void CAFMaker::FillBeamInfo(caf::SRBeamBranch &beam, const art::Event &evt) const
+  {
+    //TODO: See how to retrieve these info
+    beam.ismc = true; //Hardcoded to true at the moment.
+  }
+
+  //------------------------------------------------------------------------------
 
-    auto pidin = evt.getHandle<dunemva::MVASelectPID>(fMVASelectLabel);
-    auto pidinnue = evt.getHandle<dunemva::MVASelectPID>(fMVASelectNueLabel);
-    auto pidinnumu = evt.getHandle<dunemva::MVASelectPID>(fMVASelectNumuLabel);
+  void CAFMaker::FillCVNInfo(caf::SRCVNScoreBranch &cvnBranch, const art::Event &evt) const
+  {
     art::InputTag itag1(fCVNLabel, "cvnresult");
```

Comment:
To prevent hardcoding module names, you could just remove this and put in the fcl `CVNLabel: cvneval:cvnresult`